x11-misc/dex-0.10.1 added

kickhead13 opened the pull request:
x11-misc/dex was a couple of versions behind, now it is up-to-date.

I would love to be a maintainer for this package, it's pretty basic and I think I can handle it. :)

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

juippis commented on x11-misc/dex/dex-0.10.1.ebuild:
```diff
@@ -9,14 +9,13 @@ inherit python-r1
 KEYWORDS="amd64 x86 ~amd64-linux ~x86-linux"
 IUSE="doc"
 REQUIRED_USE="${PYTHON_REQUIRED_USE}"
-# https://github.com/jceb/dex/issues/37
```
No no, we want to keep a link to an upstream issue about why the tests are disabled for this package. Actually, I see there's a PR link in that issue. Could you try applying the patch from that PR and see if it fixes the tests, allowing us to unrestrict it in the ebuild?

kickhead13 replied:
The PR patch seems to have the same behavior as the installed version from the new ebuild. You can see this here: https://cdn.discordapp.com/attachments/1001236756814897314/1345469767645139115/image.png?ex=67c4a9de&is=67c3585e&hm=8acf428934fc32e99d7e1eaeafabe7b2f99be672048f19cd700fd390a642ea57& (the photo shows the head of the PR patch, followed by --test and --version runs of the PR patch and ebuild versions respectively).

Should I make an ebuild for the patch itself? It seems weird to do so since it isn't part of any release from the official dex repo. I'm asking this since I didn't really understand your request.

juippis replied:
Sorry, I don't understand the question. You basically take the patch file from this PR:
https://github.com/jceb/dex/pull/74

and add the .patch to this ebuild, via `${FILESDIR}` and `PATCHES` array in the ebuild.

juippis commented on x11-misc/dex/dex-0.10.1.ebuild:
```diff
@@ -9,14 +9,13 @@ inherit python-r1
 
 LICENSE="GPL-3"
 SLOT="0"
 KEYWORDS="amd64 x86 ~amd64-linux ~x86-linux"
```
Always drop keywords back to testing when adding a new version.
**ekeyword ~all dex-0.10.1.ebuild**

juippis reviewed (commented):
You can't drop the stable version in a same commit where you introduce new version. You must retain the stable version while adding a completely new version with "testing" keywords. 

You can fix the commit in this PR, see how to do git force push.

Could you also try adding python-3.13 compatibility to the package, and 3.10 can be dropped.

kickhead13 commented:
Ok! I'll be fixing this after ai get off work later today.

Brb.

kickhead13 commented:
Fixed issues (added stable version back, added upstream issue link, dropped keywords to testing, added python 3.13 support and dropped 3.10 support for the newer version).
